### ISSUE TITLE
SetStatus (int sc, java.lang.String sm) of HttpServletResponse is deprecated

### ISSUE THREAD

**Issue by magnolia-k** [scalatra/scalatra#709]
```
[warn] /Users/xxx/scalatra/core/src/main/scala/org/scalatra/servlet/RichResponse.scala:21: method setStatus in trait HttpServletResponse is deprecated: see corresponding Javadoc for more information.
[warn]     res.setStatus(statusLine.code, statusLine.message)
[warn]         ^
```

SetStatus with two arguments is deprecated.

If resutl code is 4xx or 5xx, if you want to set status line, you should change body so that it does not send body, but directly returns result using sendError method.

However, because of the HTTP protocol standard, there is no such limitation in the status line, so I do not know if it is correct to be deprecated...

**Comment by magnolia-k** [comment]
It is not inconsistent with the existing test code in the following changes

core/src/main/scala/org/scalatra/ScalatraBase.scala

-          response.status = ResponseStatus(status, reason)
+          response.sendError(status, reason)

core/src/main/scala/org/scalatra/servlet/RichResponse.scala

-    res.setStatus(statusLine.code, statusLine.message)
+    res.setStatus(statusLine.code)

**Comment by takezoe** [comment]
Fixed in #732.
